### ISSUE TITLE
CLI needs more support for client credential work flows

### ISSUE THREAD

**Issue by staylor14** [cloudfoundry/cli#1418]
## Command

When we authenticate using client credentials, using a client with the `cloud_controller.admin` authority, we're unable to create orgs as expected.

```
# What I'd like to do:
cf auth client_id client_secret --client-credentials
cf create-org myorg
cf target -o myorg
cf create-space myspace
```

## What occurred

```
[2018-07-11 19:03:15.74 (UTC)]> cf create-org SMOKE-1-ORG-601c8de8f6100d85
Creating org SMOKE-1-ORG-601c8de8f6100d85 as ...
OK

Assigning role OrgManager to user  in org SMOKE-1-ORG-601c8de8f6100d85 ...
FAILED
Failed assigning org role to user: Server error, status code: 404, error code: 20003, message: The user could not be found:
```

## What you expected to occur

I expected CF CLI to have exit status 0 after creating the org. Assigning the OrgManager role is unexpected.

## CLI Version

cf version 6.37.0+a40009753.2018-05-25

## CC API Endpoint Version

2.115.0

## CF Trace

nah

## Platform & Shell Details

nah

## Any other relevant information

We need this to make progress on [this security-related feature narrative](https://docs.google.com/document/d/1mtluLgqxtLDIRZnF2XHXke4DSiJhx5roMs7PEdv4VCE/). I'm working on this with @jhamon from UAA team.

**Comment by abbyachau** [comment]
Thanks @staylor14, the CF CLI will need to refactor `create-org` before we can dig into this bug. I've put this story in the icebox until the [refactor](https://www.pivotaltracker.com/story/show/126256673) is done.

**Comment by abbyachau** [comment]
Note this the `create-org` bug was fixed in [CLI v6.39.0](https://github.com/cloudfoundry/cli/releases/tag/v6.39.0). However we are leaving this issue open for since we have not implemented the same fix for `create-space`.

**Comment by XenoPhex** [comment]
This issue has been resolved for the `create-org` and `create-space` commands in [cf V6.40.0](https://github.com/cloudfoundry/cli/releases/tag/v6.40.0).

If anyone happens to stumble upon other commands that do not work with `--client-credentials`, please file additional issues. Thanks everyone!
